### PR TITLE
Add AccountRegistry

### PR DESCRIPTION
**Pull request by TobiasFella**
Basic session management class; Created from Quaternion's AccountRegistry and NeoChat's AccountListModel.
The connections are stored in a map userId->Connection to facilitate session lookup; This limits it to one connection for each account, which shouldn't be a problem realistically. I could also change it to something like (userId, deviceId)->connection

### CODE REVIEW

File: lib/accountregistry.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    void drop(Connection* a);
+    bool isLoggedIn(const QString& userId) const;
+    Connection* operator[](int);
+    Connection* operator[](int) const;
+    bool isEmpty() const;
+    int count() const;
+    bool contains(Connection*) const;
+    Connection* get(const QString& userId);
+
+    const_iterator cbegin() const;
```
Comment (CarlSchwan) [inline lib/accountregistry.h:39]: This is very dangerous. See this fun thread https://mail.kde.org/pipermail/kde-pim/2021-April/047417.html

Tldr; this is not abi compatible if quotient is compiled with some flags but neochat/quatermion with another flag

**Review by CarlSchwan — COMMENTED** [review]
Looks good to me, other than the iterator issue  :/

File: lib/accountregistry.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+class AccountRegistry : public QAbstractListModel {
+    Q_OBJECT
+public:
+    using const_iterator = QMap<QString, Connection *>::const_iterator;
+
+    enum EventRoles {
+        ConnectionRole = Qt::UserRole + 1,
+    };
+
+    static AccountRegistry &instance() {
```
Comment (KitsuneRal) [inline lib/accountregistry.h:23]: `Q_GLOBAL_STATIC`? Though I'm not totally sure how much one is better than another in this case.

File: lib/accountregistry.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    [[nodiscard]] QHash<int, QByteArray> roleNames() const override;
+
+private:
+    AccountRegistry();
+
+    QMap<QString, Connection *> m_accounts;
+
+signals:
+    void addedAccount(Connection* a);
+    void aboutToDropAccount(Connection* a);
```
Comment (KitsuneRal) [inline lib/accountregistry.h:56]: As you derive from `QAbstractListModel`, I don't think it's worth replicating signals that serve the same function (even if they provide less simplistic API). Quaternion will do fine with connecting to `rowsInserted()` and `aboutToRemoveRows()` instead.

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
